includes/admin/class-wp-job-manager-promoted-jobs-admin.php

```diff
@@ -390,6 +391,52 @@ public function maybe_add_promoted_jobs_notice( $notices ) {
 
 		return $notices;
 	}
+
+	/**
+	 * Add a notice to the job listing admin page if there are promoted jobs on trash.
+	 *
+	 * @internal
+	 *
+	 * @param array $notices Notices to filter on.
+	 *
+	 * @return array
+	 */
+	public function maybe_add_trash_notice( $notices ) {
+		$promoted_trash_count = WP_Job_Manager_Promoted_Jobs::query_promoted_jobs_count( [ 'post_status' => 'trash' ] );
+		if ( 0 === $promoted_trash_count ) {
+			return $notices;
+		}
+
+		$trash_url = add_query_arg(
+			[
+				'post_type'   => 'job_listing',
+				'post_status' => 'trash',
+			],
+			admin_url( 'edit.php' )
+		);
+
+		$notices['promoted-job-on-trash'] = [
+			'type'        => 'user',
+			'dismissible' => false,
```

Comment:
I made it not dismissable, so we make sure it will show up every time the user has a promoted job on trash.